components/raftstore/src/store/fsm/peer.rs

```diff
@@ -778,6 +782,9 @@ where
             }
             self.fsm.batch_req_builder.request = Some(cmd);
         }
+        if self.fsm.batch_req_builder.request.is_some() {
+            self.ctx.raft_metrics.ready.propose_delay.inc();
```

Comment:
Maybe we should place this line in the else branch above for better readability?